src/lib/extensions.ts

```diff
@@ -0,0 +1,27 @@
+import { OptionValues } from 'commander'
+import isEmpty from 'lodash/isEmpty.js'
+import BaseCommand from '../commands/base-command.js'
+import { init } from '../commands/init/init.js'
+import { log } from '../utils/command-helpers.js'
+
+export const packagesThatNeedSites = ['@netlify/neon']
+
+export async function handleExtensionRequirements(options: OptionValues, command: BaseCommand) {
```

Comment:
I would instead refactor this function to `doesProjectRequireLinkedSite` that returns a boolean. It'd accept a `project`, a `site`, and a `siteInfo`. You'd then move the `await init(...)` call out to `dev.ts`.